katharsis-client/src/main/java/io/katharsis/client/internal/AbstractStub.java

```diff
@@ -0,0 +1,96 @@
+package io.katharsis.client.internal;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.squareup.okhttp.HttpUrl;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Request.Builder;
+import com.squareup.okhttp.Response;
+
+import io.katharsis.client.ClientException;
+import io.katharsis.client.KatharsisClient;
+import io.katharsis.errorhandling.ErrorData;
+import io.katharsis.errorhandling.ErrorResponse;
+import io.katharsis.errorhandling.mapper.ExceptionMapper;
+import io.katharsis.errorhandling.mapper.ExceptionMapperRegistry;
+import io.katharsis.response.BaseResponseContext;
+import io.katharsis.utils.java.Optional;
+
+public class AbstractStub {
+
+	// TODO katharsis constant
+	private static final String CONTENT_TYPE = "application/vnd.api+json";
+
+	protected KatharsisClient katharsis;
+	protected RequestUrlBuilder urlBuilder;
+
+	public AbstractStub(KatharsisClient client, RequestUrlBuilder urlBuilder) {
+		this.katharsis = client;
+		this.urlBuilder = urlBuilder;
+	}
+
+	protected BaseResponseContext executeGet(HttpUrl requestUrl) {
+		Builder builder = new Request.Builder().url(requestUrl);
+		return execute(builder, true);
+	}
+
+	protected BaseResponseContext executeDelete(HttpUrl requestUrl) {
+		Builder builder = new Request.Builder().url(requestUrl);
+		builder.delete();
+		return execute(builder, false);
+	}
+
+	protected BaseResponseContext execute(Builder builder, boolean getResponse) {
+		try {
+			builder = builder.header("Content-Type", CONTENT_TYPE);
+
+			Request request = builder.build();
+			Response response = katharsis.getHttpClient().newCall(request).execute();
+			if (!response.isSuccessful()) {
+				handleError(response);
+			}
+
+			String body = response.body().string();
+			ObjectMapper objectMapper = katharsis.getObjectMapper();
+			if (getResponse) {
+				return objectMapper.readValue(body, BaseResponseContext.class);
+			} else {
+				return null;
+			}
+		} catch (IOException e) {
+			throw new IllegalStateException(e);
+		}
+	}
+
+	@SuppressWarnings({ "unchecked" })
+	private void handleError(Response response) throws JsonParseException, JsonMappingException, IOException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'com.fasterxml.jackson.core.JsonParseException' which is a subclass of 'java.io.IOException'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'com.fasterxml.jackson.databind.JsonMappingException' which is a subclass of 'java.io.IOException'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)